src/api/common/HttpClient.ts

```diff
@@ -0,0 +1,73 @@
+import log from 'loglevel';
+import type {
+    HubInfo,
+    RelayEstimation,
+    RelayOrDeployRequest
+} from 'src/common/relay.types';
+import type HttpWrapper from './HttpWrapper';
+
+const PATHS = {
+  GET_INFO: '/getaddr',
+  POST_REALY_REQEUST: '/relay',
+  POST_ESTIMATE: '/estimate'
+} as const;
+const VERIFIER_SUFFIX = '?verifier=';
+
+export default class HttpClient {
+  private readonly _httpWrapper: HttpWrapper;
+
+  constructor(httpWrapper: HttpWrapper) {
+    this._httpWrapper = httpWrapper;
+  }
+
+  async getHubInfo(relayUrl: string, verifier = ''): Promise<HubInfo> {
+    const verifierSuffix = verifier && `${VERIFIER_SUFFIX}${verifier}`;
+    const hubInfo: HubInfo = await this._httpWrapper.sendPromise(
+      relayUrl + PATHS.GET_INFO + verifierSuffix
+    );
+    if (!hubInfo) {
+      throw new Error('Relay responded without a body');
+    }
+    log.info(`hubInfo: ${JSON.stringify(hubInfo)}`);
+
+    return hubInfo;
+  }
+
+  async relayTransaction(
+    relayUrl: string,
+    request: RelayOrDeployRequest
+  ): Promise<string> {
+    // TODO: where does the response type come from?
+    const { signedTx, error } = await this._httpWrapper.sendPromise<{
+      signedTx: string;
+      error: string;
+    }>(relayUrl + PATHS.POST_REALY_REQEUST, request);
+    log.info('relayTransaction response:', signedTx, error);
+    if (error) {
+      throw new Error(`Got error response from relay: ${error}`);
+    }
```

Comment:
Probably we could create a method to handle the error, since we're repeating the same code here and in the `estimateMaxPossibleGas`

Reply:
actually, I reckon this should be an interceptor.